bindings/nodejs/lib/types/block/address.ts

```diff
@@ -118,12 +133,105 @@ class NftAddress extends Address {
     }
 }
 
+/**
+ * An implicit account creation address.
+ */
+class ImplicitAccountCreationAddress extends Address {
+    /**
+     * The Ed25519 address.
+     */
+    private pubKeyHash: HexEncodedString;
+    /**
+     * @param address An Ed25519 address.
+     */
+    constructor(address: Ed25519Address) {
+        super(AddressType.ImplicitAccountCreation);
+        this.pubKeyHash = address.pubKeyHash;
+    }
+
+    address(): Ed25519Address {
+        return new Ed25519Address(this.pubKeyHash);
+    }
+
+    toString(): string {
+        return this.address.toString();
+    }
+}
+
+const RestrictedAddressDiscriminator = {
+    property: 'type',
+    subTypes: [
+        { value: Ed25519Address, name: AddressType.Ed25519 as any },
+        { value: AccountAddress, name: AddressType.Account as any },
+        { value: NftAddress, name: AddressType.Nft as any },
+    ],
+};
+
+/**
+ * An address with restricted capabilities.
+ */
+class RestrictedAddress extends Address {
+    /**
+     * The inner address.
+     */
+    @Type(() => Address, {
+        discriminator: RestrictedAddressDiscriminator,
+    })
+    readonly address: Address;
+    /**
+     * The allowed capabilities bitflags.
+     */
+    private allowedCapabilities: HexEncodedString = '0x00';
+    /**
+     * @param address An address.
+     */
+    constructor(address: Address) {
+        super(AddressType.Restricted);
+        this.address = address;
+    }
+
+    setAllowedCapabilities(allowedCapabilities: Uint8Array) {
+        if (allowedCapabilities.some((c) => c != 0)) {
+            this.allowedCapabilities =
+                '0x' +
+                allowedCapabilities.length.toString(16) +
+                Buffer.from(
+                    allowedCapabilities.buffer,
+                    allowedCapabilities.byteOffset,
+                    allowedCapabilities.byteLength,
+                ).toString('hex');
+        } else {
+            this.allowedCapabilities = '0x00';
+        }
+    }
+
+    withAllowedCapabilities(
+        allowedCapabilities: Uint8Array,
+    ): RestrictedAddress {
+        this.setAllowedCapabilities(allowedCapabilities);
+        return this;
+    }
+
+    getAllowedCapabilities(): Uint8Array {
+        return Uint8Array.from(Buffer.from(this.allowedCapabilities, 'hex'));
+    }
+
+    toString(): string {
+        return this.address.toString() + this.allowedCapabilities.substring(2);
```

Comment:
Just making sure, this is never used by any JSON utility, right?

Reply:
I don't think so, but not sure